src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -683,6 +683,56 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
             return true; // to skip internal onPaste handler
         }
 
+        // Safari `Insert from iPhone or iPad`
+        // data.getData("text/html") returns a string like: <img src="blob:https://...">
+        if (data.types.includes("text/html")) {
+            const imgElementStr = data.getData("text/html");
+            const parser = new DOMParser();
+            const imgDoc = parser.parseFromString(imgElementStr, "text/html");
+
+            if (
+                imgDoc.querySelector("img") == null ||
+                imgDoc.querySelector("img")!.src == null ||
+                imgDoc.getElementsByTagName("img").length != 1 ||
+                !imgDoc.querySelector("img")!.src.startsWith("blob:")
```

Comment:
```suggestion
                imgDoc.getElementsByTagName("img").length !== 1 ||
                !imgDoc.querySelector("img")?.src.startsWith("blob:")
```

Reply:
We should also be checking that the img is the only node in the DOM, no text nodes, divs, etc that could be after the single img node (i.e. if someone copied an img + caption from somewhere) - SlashCommands.tsx `addwidget`'s iframe handling is likely to be a good example